open-craft/sprints
# [BB-1621] Prepare frontend for billable/non-billable hours

This introduces the frontend for the sustainability dashboard.
I made a change to the original design by removing the `select` box for choosing overall/cell/person stats, because using it was very unintuitive. Instead, the sustainability dashboard is aware of the sprint board's current view (whether it's showing cells/cell's board/person's board). Therefore, when you click on the cell's name, the sustainability dashboard recalculates its data for displaying cell-related data only.

All calculations here are performed on the frontend side and stored in the Local Storage, so the navigation is smooth. To reload the sustainability dashboard you need to refresh the whole page.

## Testing instructions:
1. Set `.env` file.
1. Start the backend with `docker-compose -f local.yml up --build`.
1. Create frontend/.env.local and set REACT_APP_GOOGLE_CLIENT_ID there.
1. Navigate to frontend and run npm i && npm start.
1. Log in.
1. Check the Sustainability dashboard for overall/cells/yourself and see if the data look good.
1. Change `From` and `To` dates and recheck one of the cases mentioned above.
